testsuite/generate-format/tests/detect_format_change.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use generate_format::{add_proptest_serialization_tracing, FILE_PATH};
+use serde_reflection::{RegistryOwned, Tracer};
+use serde_yaml;
+use std::collections::BTreeMap;
+
+static MESSAGE: &str = "Run `cargo run -p generate-format -- --record` to refresh the records and `git diff` to study the changes before amending your commit.";
+
+#[test]
+fn test_format_change() {
```

Comment:
could you add a negative test? although maybe that's just the same as verifying two yamls do not match... but.. thoughts?

Reply:
Interesting. I'm trying to come up with a scenario where the test would stop working later on, and I don't find it (except someone overriding PartialEq on Formats for no reason :)), so maybe that's fine.